Yadda - Part 1

paulh16 opened the pull request:
Please find the shema for Part 1 of Yadda. A couple notes:

1. I defined a schema for this SQL file: yadda.

2. I have added comments in the file explaining the relationships between tables.

3. The primary key for the rating table is a composite key (user and beer).

Thank you!

incazteca commented on yadda/yadda-pg.sql:
```diff
@@ -0,0 +1,84 @@
+
+-- Role Table
+DROP TABLE IF EXISTS yadda.role CASCADE;
+CREATE TABLE yadda.role (
```
tables should be in plural. For example, table role should be roles.

incazteca commented on yadda/yadda-pg.sql:
```diff
@@ -0,0 +1,84 @@
+
+
+-- User Table
+DROP TABLE IF EXISTS yadda.user CASCADE;
```
Pluralize table name

incazteca commented on yadda/yadda-pg.sql:
```diff
@@ -0,0 +1,84 @@
+  active        boolean DEFAULT true,
+  created_at    timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  updated_at    timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  role_id       int REFERENCES yadda.role ON DELETE RESTRICT  -- one-to-many relationship between role and users; disallow deletion of role if it still references a user
```
What exactly is the role table for? I didn't see it in the requirements.

incazteca commented on yadda/yadda-pg.sql:
```diff
@@ -0,0 +1,84 @@
+CREATE TABLE yadda.beer (
+  id            serial PRIMARY KEY,
+  name          text NOT NULL,
+  style         text NOT NULL,
```
Consider using a lookup table for the beer styles.

incazteca commented on yadda/yadda-pg.sql:
```diff
@@ -0,0 +1,84 @@
+  style         text NOT NULL,
+  description   text,
+  brewing_year  date,
+  created_at    timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP,
```
Ensure that your timestamps have a timezone with them. Use `timestamptz` to keep the timezone.

incazteca commented on yadda/yadda-pg.sql:
```diff
@@ -0,0 +1,84 @@
+  phone_number  varchar(30) NULL,
+  email         varchar(500) NOT NULL,
+  active        boolean DEFAULT true,
+  created_at    timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP,
```
Missing updated_by and created_by in this table. Actually @jmmastey and @djkotowski Are we wanting to know what database user had updated this record? Or is it the user/customer that we will be tracking.

incazteca commented on yadda/yadda-pg.sql:
```diff
@@ -0,0 +1,84 @@
+  smell         int CHECK (smell BETWEEN 0 AND 5) DEFAULT 0,
+  taste         int CHECK (taste BETWEEN 0 AND 5) DEFAULT 0,
+  feel          int CHECK (feel BETWEEN 0 AND 5) DEFAULT 0,
+  overall       int CHECK (overall BETWEEN 0 AND 5) DEFAULT 0,
```
Should overall be an average of the other criteria? Actually that's a nitpick that doesn't really matter. Sorry.

incazteca commented on yadda/yadda-pg.sql:
```diff
@@ -0,0 +1,84 @@
+  state         varchar(2) NOT NULL,
+  zip_code      varchar(5) NOT NULL,
+  description   text,
+  founding_year date,
```
I don't think a date is exactly needed in this case since the column name is year. You could just have an integer. However if you are using a date you should probably change the column name to `founding_date`.

incazteca commented on yadda/yadda-pg.sql:
```diff
@@ -0,0 +1,84 @@
+
+-- Rating Table
+DROP TABLE IF EXISTS yadda.rating CASCADE;
+CREATE TABLE yadda.rating (
```
The ratings table itself should have an id column.

incazteca commented:
Comments for Part 1: Some small changes could be made however biggest point is probably normalizing the beer table and ensuring that the table names are pluralized. I will expect some indexes to also be created for the foreign keys however, I think that was saved for part 4. Please keep this PR open as you add the other parts.

paulh16 commented:
I updated the file per the recommendations by @incazteca.

I also deleted the reference to the schema, which isn't needed for this exercise.

Thanks!

djkotowski commented:
Please open new PRs for each review.
